### ISSUE TITLE
ping function call returns PONG when Redis server is not running (sock connection).

### ISSUE THREAD

**Issue by dstruck** [aio-libs/aioredis-py#773]
```
import aioredis
conn = await aioredis.create_redis_pool('unix:///var/run/redis/redis-server.sock?db=0')
await conn.ping()
```

returns **b'PONG'** even though the server is not running.

```
import redis
r = redis.Redis('unix:///var/run/redis/redis-server.sock?db=0')
r.ping()
```

raises the **ConnectionError** exception.

aioredis should not return **b'PONG'** on its own, it should only come from a Redis server.

Maybe `self._process_data(data or b'PONG')` (aioredis/connection.py) is the culprit?

**Comment by seandstewart** [comment]
@dstruck -

Thanks for the ticket and the initial investigation! Would you be willing to submit a PR with a potential fix for this problem?

**Comment by miketzian** [comment]
Might this be already fixed, or had some other cause?

I get FileNotFoundError when redis isn't running and ConnectionRefusedError when I try to fake the file (was going to look at fixing this)

**Comment by seandstewart** [comment]
@miketzian -

It has not - would welcome a PR with a fix if you have time.

**Comment by miketzian** [comment]
I'd be happy to, though I can't reproduce it. When redis isn't running, the socket file doesn't exist and I get a FileNotFoundError and not b'PONG'

**Comment by waketzheng** [comment]
My machine is Mac with aioredis1.3 and got FileNotFoundError either. @dstruck how to reproduct that `pong`?

**Comment by seandstewart** [comment]
This issue should be resolved as of #891 - please feel free to pull in the latest master and test to ensure this is the case.
